src/m365/spo/commands/group/group-user-add.ts

```diff
@@ -33,52 +34,95 @@ class SpoGroupUserAddCommand extends SpoCommand {
   }
 
   public commandAction(logger: Logger, args: CommandArgs, cb: (err?: any) => void): void {
-    this.getOnlyActiveUsers(args, logger)
-      .then((resolvedUsernameList: string[]): Promise<SharingResult> => {
-        if (this.verbose) {
-          logger.logToStderr(`Start adding Active user/s to SharePoint Group ${args.options.groupId}...`);
-        }
-
-        const data: any = {
-          url: args.options.webUrl,
-          peoplePickerInput: this.getFormattedUserList(resolvedUsernameList),
-          roleValue: `group:${args.options.groupId}`
-        };
-        const requestOptions: any = {
-          url: `${args.options.webUrl}/_api/SP.Web.ShareObject`,
-          headers: {
-            'Accept': 'application/json;odata=nometadata',
-            'Content-type': 'application/json;odata=verbose'
-          },
-          data: data,
-          responseType: 'json'
-        };
-
-        return request.post<SharingResult>(requestOptions);
+    this
+      .getGroupId(args)
+      .then((groupId: number): Promise<SharingResult> => {
+        return this.getOnlyActiveUsers(args, logger)
+          .then((resolvedUsernameList: string[]): Promise<SharingResult> => {
```

Comment:
Let's not nest promises because it makes code hard to read. Instead, let's flatten the structure by changing code to `return this.getOnlyActiveUsers(args, logger);` and then moving the next `then` clause to the next chain.